examples/data-objects/inventory-app/src/view/inventoryList.tsx

```diff
@@ -4,20 +4,36 @@
  */
 
 import * as React from "react";
-import { useTreeContext } from "@fluid-experimental/tree-react-api";
-import { Inventory, InventoryField } from "../schema";
+import { node } from "@fluid-experimental/tree2";
+import { Inventory } from "../schema";
 import { Counter } from "./counter";
 
-export const MainView: React.FC<{ tree: InventoryField }> = ({ tree }) => {
-	// TODO: offer an API to subscribe to invalidation from a field to avoid depending on the whole document here.
-	useTreeContext(tree.context);
-	const inventory: Inventory = tree.content;
+export const MainView: React.FC<{ inventory: Inventory }> = ({ inventory }) => {
+	// Use a React effect hook to invalidate this component when the inventory changes.
+	// We do this by incrementing a counter, which is passed as a dependency to the effect hook.
+	const [invalidations, setInvalidations] = React.useState(0);
+
+	// React effect hook that increments the 'invalidation' counter whenever inventory or any of its children change.
+	React.useEffect(() => {
+		// Returns the cleanup function to be invoked when the component unmounts.
+		return node(inventory).on("subtreeChanging", () => {
+			// TODO: RAF required because 'subtreeChanging' event fires prior to applying changes.
+			//       Remove RAF when we have an "afterChange" event.
+			requestAnimationFrame(() => setInvalidations((i) => i + 1));
+		});
+	}, [invalidations, inventory]);
```

Comment:
Inlined the state/effect hook because, at this point, it's just idiomatic React.